extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/serialization/JsonbSerializer.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkus.kafka.client.serialization;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Map;
+
+import javax.json.bind.Jsonb;
+import javax.json.bind.JsonbBuilder;
+
+import org.apache.kafka.common.serialization.Serializer;
+
+/**
+ * A {@link Serializer} that serializes to JSON using JSON-B.
+ */
+public class JsonbSerializer<U> implements Serializer<U> {
+
+    private final Jsonb jsonb;
+
+    public JsonbSerializer() {
+        jsonb = JsonbBuilder.create();
+    }
+
+    public JsonbSerializer(Jsonb jsonb) {
+        this.jsonb = jsonb;
+    }
+
+    @Override
+    public void configure(Map<String, ?> configs, boolean isKey) {
+    }
+
+    @Override
+    public byte[] serialize(String topic, U data) {
+        try (ByteArrayOutputStream output = new ByteArrayOutputStream()) {
+            jsonb.toJson(data, output);
+            return output.toByteArray();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void close() {
```

Comment:
should we close the jsonb object here? Probably only if created in this class.

Reply:
Yes, changed it into that.